cuqi/testproblem/_testproblem.py

```diff
@@ -148,69 +152,106 @@ class Deconvolution1D(BayesianProblem):
     :math:`\mathbf{x}` is a sharp (clean) signal and
     :math:`\mathbf{A}` is a convolution operator.
 
+    The convolution operator is defined by specifying a point spread function and
+    boundary conditions and is computed (matrix-free) via scipy.ndimage.convolve1D,
+    or using a matrix representation if the use-matrix flag is set to True.
```

Comment:
I can't see in the comments how to pass use-matrix flag, and if so how to pass the matrix?  How to handle if the user attempts passing both a PSF and a matrix, is that possible?  One of them should take precedence, and it should be made clear to the user.  Alternatively, could either PSF or matrix be passed as the same input (currently kernel or PSF) and then depending on type call convolve or apply matrix? Not sure if a good idea, just a thought.

Reply:
Was a mistake. Updated.